src/main/java/net/dv8tion/jda/api/interactions/components/text/TextInput.java

```diff
@@ -0,0 +1,488 @@
+         * @throws IllegalArgumentException
+         *         If the provided placeholder is longer than {@link #PLACEHOLDER_MAX_LENGTH} characters
+         *
+         * @return The same builder instance for chaining
+         */
+        @Nonnull
+        public Builder setPlaceholder(@Nullable String placeholder)
+        {
+            if (placeholder != null)
+                Checks.notLonger(placeholder, PLACEHOLDER_MAX_LENGTH, "Placeholder");
+            this.placeholder = placeholder;
+            return this;
+        }
+
+        /**
+         * The minimum length
+         *
+         * @return Minimum length
+         */
+        public int getMinLength()
+        {
+            return minLength;
+        }
+
+        /**
+         * The maximum length
+         *
+         * @return Maximum length
+         */
+        public int getMaxLength()
+        {
+            return maxLength;
+        }
+
+        /**
+         * The custom id
+         *
+         * @return Custom id
+         */
+        @Nonnull
+        public String getId()
+        {
+            return id;
+        }
+
+        /**
+         * The label shown above this text input box
+         *
+         * @return Label for the input
+         */
+        @Nonnull
+        public String getLabel()
+        {
+            return label;
+        }
+
+        /**
+         * The {@link TextInputStyle TextInputStyle}
+         *
+         * @return The TextInputStyle
+         */
+        @Nonnull
+        public TextInputStyle getStyle()
+        {
+            return style;
+        }
+
+        /**
+         * The placeholder of this TextInput
+         * <br>This is the short hint that describes the expected value of the TextInput field.
+         *
+         * @return Placeholder
+         */
+        @Nullable
+        public String getPlaceholder()
+        {
+            return placeholder;
+        }
+
+        /**
+         * The String value of this TextInput
+         *
+         * @return Value
+         */
+        @Nullable
+        public String getValue()
+        {
+            return value;
+        }
+
+        /**
+         * Whether this TextInput is required.
+         * <br>If this is True, the user must populate this TextInput field before they can submit the Modal.
+         *
+         * @return True if this TextInput is required
+         * 
+         * @see    TextInput#isRequired()
+         */
+        public boolean isRequired()
+        {
+            return required;
+        }
+
+        /**
+         * Builds a new TextInput from this Builder
+         *
+         * @throws IllegalArgumentException
+         *         If maxLength is smaller than minLength
+         *
+         * @return the TextInput instance
+         */
+        @Nonnull
+        public TextInput build()
+        {
+            if (maxLength < minLength)
+                throw new IllegalArgumentException("maxLength cannot be smaller than minLength!");
```

Comment:
If I do `.setMinLength(5)` and the `.build()` this will break because default for `maxLength` is `-1`